### PR TITLE
add noscript ui

### LINKED ISSUE #774: Test Pilot should have a simple <noscript> experience.
I can understand using Javascript for complicated web apps that need to do a lot of things, but in this case you're using JS-only, with no fallback whatsoever, to display what's at heart a completely static site with a gif.

### PR DESCRIPTION
**Pull request by johngruen**
Fixes #774

### CODE REVIEW

File: locales/en-US/app.l20n [hunk 1]
```diff
@@ -95,3 +95,7 @@
 <tourDoneButton "Done">
 <userCountContainer "There are <span class=\"bold\">{{installation_count}}</span> people trying {{title}} right now!">
 <highlightPrivacy "Your privacy">
+
+<noScriptHeading "Uh oh...">
```
Comment (lmorchard) [inline locales/en-US/app.l20n:103]: FWIW, L20N as we currently use it requires JS. So, these strings won't get touched in the noscript case.

There _is_ a [python-l20n](https://github.com/l20n/python-l20n) that seems current. Probably worth a follow-up issue to research how we might be able to use that for server-generated pages like this.

Comment (johngruen) [inline locales/en-US/app.l20n:103]: @lmorchard you want me to back that stuff out or leave it for now and file a follow up bug?

File: docs/FAQs.md [hunk 2]
```diff
@@ -11,3 +11,11 @@ As this is an evolving product, this requirement may be adjusted in the future,
...
 - We also use the account as a gate since we are doing certain metrics collections that we want to ensure people have knowingly elected to opt into.
 
+
+### Why does Test Pilot require Javascript?
+
+- We use Javascript to render the user interface of Test Pilot.
+- We use Javascript to manage experiment installations.
+- We use Javascript to collect data to help us improve Test Pilot in accordnace with our [Terms of Use](https://testpilot.firefox.com/terms) and [Privacy Notice](https://testpilot.firefox.com/privacy).
+
+Since Test Pilot is an open source project, you can see all of the cool ways we use Javascript by [examining our code](https://github.com/mozilla/testpilot/).
```
Comment (pdehaan) [inline docs/FAQs.md]: Nit: technically it's "JavaScript" (upper S)
Source: Us. We invented it. https://developer.mozilla.org/en-US/docs/Web/JavaScript

File: docs/FAQs.md [hunk 3]
```diff
@@ -11,3 +11,11 @@ As this is an evolving product, this requirement may be adjusted in the future,
 - Test Pilot is about building a community through FxA enabled forums such as Discourse.
 - We also use the account as a gate since we are doing certain metrics collections that we want to ensure people have knowingly elected to opt into.
 
+
+### Why does Test Pilot require Javascript?
+
+- We use Javascript to render the user interface of Test Pilot.
+- We use Javascript to manage experiment installations.
+- We use Javascript to collect data to help us improve Test Pilot in accordnace with our [Terms of Use](https://testpilot.firefox.com/terms) and [Privacy Notice](https://testpilot.firefox.com/privacy).
```
Comment (pdehaan) [inline docs/FAQs.md]: Typo: "accordnace" => "accordance"

File: locales/en-US/app.l20n [hunk 4]
```diff
@@ -95,3 +95,7 @@
...
+<noScriptMessage "Test Pilot requires Javascript.<br>Sorry about that.">
```
Comment (pdehaan) [inline locales/en-US/app.l20n]: "JavaScript" here too (if we leave these strings in).

File: locales/en-US/app.l20n (hunk 1 above)
Comment (lmorchard) [inline locales/en-US/app.l20n:103]: Might as well leave them in and file a follow up.

Comment (pdehaan) [inline locales/en-US/app.l20n:103]: Unpopular counter proposal: Remove l20n support for the noscript page and just serve all edge-case-users English "Test Pilot requires JavaScript" message.

I'm guessing there isn't an easy way to calculate how many people would hit the `noscript` block (since we couldn't ping GA &mdash; unless we serve up some 1px transparent beacon and see if we can check counts in HTTP logs to see how many people are browsing w/ JavaScript disabled).

Comment (lmorchard) [inline locales/en-US/app.l20n:103]: Well, I mean, that's what we're de facto doing even by leaving the L20n stuff in there. It's just that some day we might be able to support L20n on the server side, too.

**Comment by meandavejustice** [comment]
I vote we remove the `noScriptHeading` stuff since there are edits needing to be done with the pr anyways.

Other than that I am happy with the message and addition to the faqs doc. Once @lmorchard and @pdehaan 's comments are addressed I'm good to merge

File: locales/en-US/app.l20n (hunk 1 above)
Comment (johngruen) [inline locales/en-US/app.l20n:103]: How about i add a todo and we call it?

**Comment by johngruen** [comment]
@meandavejustice see my note above...my vote is to add a TODO:

**Comment by meandavejustice** [comment]
Okay, works for me, so long as we create an issue and link to it in said TODO

File: docs/FAQs.md [hunk 5]
```diff
@@ -3,11 +3,19 @@ Frequently Asked Questions
 
 ### Why does Test Pilot require a Firefox Account?
 
-Test Pilot was launched with the requirement that particpants be logged into a Firefox Account.
+Test Pilot was launched with the requirement that participants be logged into a Firefox Account.
```
Comment (johngruen) [inline docs/FAQs.md:6]: @pdehaan i see @ckprice doesn't warrant a spell check.

File: locales/en-US/app.l20n [hunk 6]
```diff
@@ -95,3 +95,11 @@
 <tourDoneButton "Done">
 <userCountContainer "There are <span class=\"bold\">{{installation_count}}</span> people trying {{title}} right now!">
 <highlightPrivacy "Your privacy">
+
+// TODO: these strings are not currently localized.
```
Comment (lmorchard) [inline locales/en-US/app.l20n]: I don't think `//` is supported in L20N format for comments, just `/* ... */` syntax.

File: docs/FAQs.md (hunk 5 above)
Comment (pdehaan) [inline docs/FAQs.md:6]: Well, @ckprice sits close enough that he can walk over and punch me, so I'm only a tough guy on The Internet when I'm confident that I'm safe.

PS: Please don't beat me up in London.

**Comment by johngruen** [comment]
@lmorchard ~~http://l20n.org/learn/comments~~ yeah, that's wrong

File: docs/FAQs.md (hunk 5 above)
Comment (johngruen) [inline docs/FAQs.md:6]: mozilla = love

**Comment by lmorchard** [comment]
No, that's FTL, the new format that [we haven't upgraded to use yet](https://github.com/mozilla/testpilot/issues/497). Old L20N format we're using [still uses `/* ... */`](https://github.com/l20n/l20n.js/blob/v1.0.x/examples/locales/en-US.l20n#L5) comments.

**Comment by johngruen** [comment]
@lmorchard i thought that seemed weird :)

**Comment by lmorchard** [comment]
Okay, I think we've beaten up on this horse enough
